Looms.PoS/Looms.PoS.Application/Features/Reservation/Commands/CreateReservation/CreateReservationRequestValidator.cs

```diff
@@ -0,0 +1,74 @@
+using FluentValidation;
+using Looms.PoS.Application.Models.Requests.Reservation;
+using Looms.PoS.Application.Utilities.Validators;
+using Looms.PoS.Application.Utilities;
+using Looms.PoS.Domain.Interfaces;
+
+namespace Looms.PoS.Application.Features.Reservation.Commands.CreateReservation;
+
+public class CreateReservationRequestValidator : AbstractValidator<CreateReservationRequest>
+{
+    public CreateReservationRequestValidator(IServicesRepository servicesRepository, IReservationsRepository reservationsRepository)
+    {
+        RuleFor(x => x.CustomerName)
+            .Cascade(CascadeMode.Stop)
+            .MustAsync(async (request, customerName, cancellation) =>
+            {
+                var appointmentTime = DateTimeHelper.ConvertToUtc(request.AppointmentTime);
+                var existingReservations = await reservationsRepository.GetReservationsByCustomerAndTimeAsync(customerName, appointmentTime);
+                var existingReservation = existingReservations.FirstOrDefault();
+                return existingReservation is null;
+            })
+            .WithMessage("An appointment for the same customer at the same time already exists.");
```

Comment:
I don't think this whole check is needed. Also, what if another customer has the same name? :D

Reply:
i mean if we cannot make the customer unique this check is useless?

Reply:
You could maybe use customer email address then? I think it already exists.

Reply:
changed that the combination of the customername and email would be unique